core/domain/classifier_services.py

```diff
@@ -214,6 +214,13 @@ def save_classifier(classifier):
         classifier_id = _create_classifier(classifier)
     else:
         _update_classifier(classifier_data_model, classifier.state_name)
+
+    # Create the mapping from <exp_id,exp_version,state_name>
+    # to classifier_id.
+    create_classifier_exploration_mapping(
```

Comment:
OK, yep -- it sounds to me like the correct course of action is to treat classifiers as write-once and have no further updates after that.

Re the last paragraph, I have a question. After the above, do we only have a create_classifier() function (and no updates)? If so, then I think it is OK for create_classifier() to call create_classifier_exploration_mapping() as a post hook, since that's the only scenario in which create_classifier_exploration_mapping() should be called. If not, then I agree that create_classifier_exploration_mapping() should be treated separately from the updating of the classifier itself. @pranavsid98 could you please advise on this?